### ISSUE TITLE
KHR_material_iridescence: missing extension support

### ISSUE THREAD

**Issue by bhouston** [ChaosGroup/vray_gltf#13]
Here are some images that demonstrate the lack of V-Ray support for [KHR_materials_iridescence](https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_materials_iridescence).

Specifically here is [khronos-IridescenceSuzanne](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/IridescenceSuzanne).

With Threejs:

![model-viewer-golden](https://github.com/ChaosGroup/vray_gltf/assets/588541/82d6714b-2754-44b8-928b-1a9def6d5cf3)

With three-gpu-pathtracer:

![three-gpu-pathtracer-golden](https://github.com/ChaosGroup/vray_gltf/assets/588541/de41b8f7-bbb7-4e69-a92a-7027b26b5b82)

And with V-Ray:

![vray-golden](https://github.com/ChaosGroup/vray_gltf/assets/588541/d8f71acf-595a-468f-9261-e22250511ba3)

**Comment by bhouston** [comment]
In V-Ray, it is called "thin film": https://docs.chaos.com/display/VMAX/VRayMtl#heading-ThinFilm

**Comment by jasondavies** [comment]
I'm going to work on this.

**Comment by jasondavies** [comment]
I'm not sure what `iridescenceFactor` and the optional`iridescenceTexture` should map to (they are combined to obtain the iridescence intensity).  Any advice?

So far, I have:

```python
iridescence_factor = o.get('iridescenceFactor', 0.0) # what should this map to?
iridescence_texture = o.get('iridescenceTexture') # what should this map to?
brtf.thin_film_ior = o.get('iridescenceIor', 1.3)
brtf.thin_film_thickness_min = o.get('iridescenceThicknessMinimum', 100.0)
brtf.thin_film_thickness_max = o.get('iridescenceThicknessMaximum', 400.0)
if o.get('iridescenceThicknessTexture') != None:
  brtf.thin_film_thickness = self._make_texture(renderer, prim, o.get('iridescenceThicknessTexture'), transfer_func=0)
```

**Comment by bhouston** [comment]
hmm... it seems that V-Ray iridescence model is more physically-based than glTF.  glTF seems to use the "factor" + "texture" as a blend onto of the iridescence model, where as V-Ray just says if you don't want iridescence, set the ior to 1 and thickness_min to 0?

I guess for now, we could add a warning to your python script if iridescenceFactor is set to something other that 0 or 1?  And also if iridescenceTexture is used at all?  If iridescenceFactor is set to 0, we should override the other parameters to disable thin_film effect.  If it is 1, we can ignore it.

Does this make sense?

**Comment by bhouston** [comment]
Maybe if iridescenceFactor is set to 0, we can disable the thin_film effect.  I believe there is a flag for this?  At least in the VRay 3DS Max UI it is a flag.  It probably is as well in the AppSDK?

**Comment by jasondavies** [comment]
Makes sense, thanks.  Yes, if the factor is `0.0` (the default), we can simply ignore the entire extension as per the spec.

**Comment by jasondavies** [comment]
Teaser shot...

![image](https://github.com/ChaosGroup/vray_gltf/assets/31223/4e2fa111-5f98-4669-a48e-0a5b14254b7e)

**Comment by bhouston** [comment]
I don't see a lot of iridescence in that image.  Maybe try [khronos-IridescenceSuzanne](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/IridescenceSuzanne) instead?

**Comment by jasondavies** [comment]
I can't render that until I implement volume support, unfortunately.  Soon!

**Comment by jasondavies** [comment]
I suppose I can render it without volume support, just to test iridescence.  Watch this space.
